src/views/Account/Rewards/PromotionSummary.tsx

```diff
@@ -0,0 +1,230 @@
+                    className='relative mr-1'
+                    chainId={chainId}
+                    address={tokenData.address}
+                    style={{ top: -2 }}
+                  />
+                  <TokenSymbol chainId={chainId} tokenData={tokenData} />
+                </>
+              )
+            }}
+          />
+        </div>
+      </div>
+
+      <div
+        className={classNames(
+          'flex items-center justify-end w-full font-averta-bold space-x-4 mt-4'
+        )}
+      >
+        <Link href={`/deposit?network=${getNetworkNiceNameByChainId(chainId).toLowerCase()}`}>
+          <a className='flex items-center h-8 uppercase text-white text-opacity-80 hover:text-opacity-100'>
+            {t('deposit')} <FeatherIcon icon='chevron-right' className={'relative w-4 h-4'} />
+          </a>
+        </Link>
+
+        {chainId === CHAIN_ID.optimism && (
+          <a
+            href={`https://app.optimism.io/bridge`}
+            target='_blank'
+            className='flex items-center h-8 uppercase text-white text-opacity-80 hover:text-opacity-100'
+          >
+            {t('bridge')}{' '}
+            <FeatherIcon
+              icon='external-link'
+              className={'relative w-4 h-4 ml-2'}
+              style={{ top: -1 }}
+            />
+          </a>
+        )}
+      </div>
+
+      {/* <BlockExplorerLink className='flex items-center' chainId={chainId} address={token} noIcon>
+          <TokenDisplay chainId={chainId} tokenData={tokenData} />
+        </BlockExplorerLink> */}
+      {/* 
+      <div className='opacity-70 text-xxs mt-2'>
+        <span className='font-bold'>Starts:</span>{' '}
+        <StartTimestampDisplay startTimestamp={startTimestamp} />
+        <br />
+        <span className='font-bold'>Ends:</span>{' '}
+        <EndTimestampDisplay
+          startTimestamp={startTimestamp}
+          numberOfEpochs={numberOfEpochs}
+          epochDuration={epochDuration}
+        />
+      </div> */}
+    </>
+  )
+}
+
+/**
+ *
+ * @param props
+ * @returns
+ */
+const StartTimestampDisplay: React.FC<{
+  startTimestamp: number
+}> = ({ startTimestamp }) => {
+  const { t } = useTranslation()
+  return (
+    <span>
+      {format(new Date(sToMs(startTimestamp)), 'MMMM do yyyy')} @{' '}
+      {format(new Date(sToMs(startTimestamp)), 'p')}
+    </span>
+  )
+}
+
+/**
+ *
+ * @param props
+ * @returns
+ */
+const EndTimestampDisplay: React.FC<{
+  startTimestamp: number
+  numberOfEpochs: number
+  epochDuration: number
+}> = ({ startTimestamp, numberOfEpochs, epochDuration }) => {
+  const { t } = useTranslation()
+
+  const duration = Number(numberOfEpochs) * Number(epochDuration)
+  const endTimestamp = Number(startTimestamp) + duration
+
+  return (
+    <span>
+      {format(new Date(sToMs(endTimestamp)), 'MMMM do yyyy')} @{' '}
+      {format(new Date(sToMs(endTimestamp)), 'p')}
+    </span>
+  )
+}
+
+export const TokenDisplay = (props) => {
```

Comment:
`TokenDisplay` and `TokenSymbol` aren't rendered if `tokenData` isn't fetched and aren't used outside of this file - these extra checks & null returns are redundant